infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInternalRuntime.java

```diff
@@ -92,7 +97,11 @@ protected void startMachines() throws InfrastructureException {
     }
     // TODO https://github.com/eclipse/che/issues/7653
     // project.pods().watch(new AbnormalStopHandler());
-    // project.pods().watchContainers(new MachineLogsPublisher());
+
+    project.pods().watchContainers(new MachineLogsPublisher());
```

Comment:
TLDR please, add finally here as well
Since we set watchers here we should also ensure that all the watchers are properly deleted. We can not guarantee that there is no bug which prevents the execution of finally block from k8s infra.

Reply:
sorry, not following - `startMachines` is supposed to be called from `internalStart` where watchers closed / deleted in finally. Could you describe potential bug that could prevent closing watchers currently ?

Reply:
Sorry, my bad. We don't need finally here.